server.go

```diff
@@ -702,11 +702,16 @@ func (s *Server) waitForReady() error {
 	// this part of the code, the called component will handle the quit
 	// signal.
 	select {
-	case <-s.ready:
-		return nil
-
 	case <-s.quit:
 		return fmt.Errorf("tapd is shutting down")
+
+	default:
+		select {
+		case <-s.ready:
```

Comment:
Maybe mention here that we now _wait_ until the server becomes ready or shuts down (meaning this _is_ blocking).